tensorflow_datasets/image/icoseg.py

```diff
@@ -0,0 +1,74 @@
+"""CMU-Cornell iCoseg dataset for image segmentation."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets.public_api as tfds
+
+_CITATION = """
+@inproceedings {
+title = {iCoseg: Interactive Co-segmentation with Intelligent Scribble Guidance},
+author = {D. Batra and A. Kowdle and D. Parikh and J. Luo and T. Chen},
+booktitle = {ECCV},
+year = {2010} }
+"""
+
+_DESCRIPTION = """\
+This is the CMU-Cornell iCoseg dataset for image segmentation. The dataset contains 38 different \
+groups totalling 643 images. These images also have their respective pixel level groundtruth \
+annotations. These groundtruths segment the images into two classes - the subject and the background.
+"""
+
+_URL = "http://chenlab.ece.cornell.edu/projects/touch-coseg/CMU_Cornell_iCoseg_dataset.zip"
+
+_HOMEPAGE_URL = "http://chenlab.ece.cornell.edu/projects/touch-coseg/"
+
+class Icoseg(tfds.core.GeneratorBasedBuilder):
+  """iCoseg dataset for segmentation."""
+
+  VERSION = tfds.core.Version('2.0.0')
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+      builder=self,
+      description=_DESCRIPTION,
+      features=tfds.features.FeaturesDict({
+            "image": tfds.features.Image(shape=(None, None, 3), encoding_format="jpeg"),
+        "label_png": tfds.features.Image(shape=(None, None, 1), encoding_format="png"),
```

Comment:
use 4 spaces here, before `"image"` and `"label_png"`. Name it as "label" instead of "label_png", to be more consistent with the other similar datasets.